### ISSUE TITLE
FirebaseApp.configure(options: FirebaseOptions) crashes app

### ISSUE THREAD

**Issue by nikolas-theodosis** [firebase/firebase-ios-sdk#1581]
* Xcode version: 9.4 (9F1027a)
  * Firebase SDK version: 5.0.0
  * Firebase Component: Core
  * Component version: 5.0.0

I am using FirebaseApp.configure(options: FirebaseOptions) since I don't want to use the GoogleService-Info.plist file. Instead, the necessary keys to configure Firebase come from my host server. If the keys received are valid there is no problem at all. However, a slight change to any key crashes the app with the following reason:

`*** Terminating app due to uncaught exception 'com.firebase.core', reason: 'Configuration fails. It may be caused by an invalid GOOGLE_APP_ID in GoogleService-Info.plist or set in the customised options.'` 

How is it possible to avoid crashing the app and instead skip Firebase configuration in that case of an error/exception?

**Comment by morganchen12** [comment]
It's currently not possible to conditionally skip Firebase configuration. You also shouldn't be modifying the keys used to configure Firebase. Waiting for a network request will delay Firebase configuration as well, which may cause you to lose some Analytics data. What's your use case for this?

**Comment by nikolas-theodosis** [comment]
I see your point but let me rephrase my question. Is it possible to catch the exceptions thrown by Firebase? I assume a delegate should be available to allow each user to choose how to handle these exceptions. Being unable to configure/initialise Firebase shouldn't lead to app crash.

**Comment by morganchen12** [comment]
This is currently unsupported by Firebase.

**Comment by nikolas-theodosis** [comment]
I see...Thank you for your help!
